engine/classes/Elgg/Http/ResponseFactory.php

```diff
@@ -144,7 +144,10 @@ public function send(Response $response) {
 						. (string) $this->response_sent);
 			}
 		} else {
-			$response->send();
+			if (elgg_trigger_before_event('send', 'http_response', $response)) {
```

Comment:
If cancelled, what should happen? Certainly the after event shouldn't be called.